cursive/examples/filters.rs

```diff
@@ -0,0 +1,135 @@
+use cursive::view::Resizable;
+use cursive::views::{Dialog, LinearLayout, RadioGroup, TextView};
+
+#[derive(Clone, Copy, Debug)]
+enum Color {
+    Red,
+    Blue,
+}
+
+#[derive(Clone, Copy, Debug)]
+enum Size {
+    Large,
+    Small,
+}
+
+#[derive(Default)]
+struct Filters {
+    color: Option<Color>,
+    size: Option<Size>,
+}
+
+// This is an examples of a Settings/Configuration "pop-up" where we can
+// modify user_data to be used elsewhere in our application.
+
+fn main() {
+    let mut siv = cursive::default();
+    siv.set_user_data(Filters::default());
+    // Leave the menu visible
+    siv.set_autohide_menu(false);
+    siv.menubar()
+        .add_leaf("Filters", |s| {
+            // Hide menu while "pop-up" settings are visible
+            s.set_autohide_menu(true);
+
+            let mut color_group: RadioGroup<Option<Color>> = RadioGroup::new();
+            let mut size_group: RadioGroup<Option<Size>> = RadioGroup::new();
+
+            // Get current filters to draw the correct selected buttons.
+            let user_data = s.user_data::<Filters>().unwrap();
+            let current_color = user_data.color;
+            let current_size = user_data.size;
```

Comment:
Note that `.user_data` keeps `Cursive` borrowed. While rust is smart enough to "drop" the borrow here just in time to let us call `.add_layer`, it may be safer to use something like `.with_user_data` instead to more clearly limit the borrow:

```suggestion
            let (current_color, current_size) = s.with_user_data(|filters: &mut Filters| {
                (filters.color, filters.size)
            };
```